src/components/vice/admin/table/actionButtons.js

```diff
@@ -0,0 +1,151 @@
+import React, { useState, useEffect } from "react";
+import { useQuery } from "react-query";
+
+import { Button, Popper } from "@material-ui/core";
+import { Skeleton } from "@material-ui/lab";
+
+import { getMessage as msg } from "@cyverse-de/ui-lib";
+
+import { id } from "./functions";
+import useStyles from "./styles";
+
+import { asyncData } from "../../../../serviceFacades/vice/admin";
+
+const ActionButtonsSkeleton = () => {
+    return (
+        <Skeleton variant="rect" animation="wave" height={75} width="100%" />
+    );
+};
+
+const ActionButton = ({ baseID, name, handler, onClick, popperMsgKey }) => {
+    const classes = useStyles();
+    return (
+        <Button
+            id={id(baseID, "button", name)}
+            variant="contained"
+            color="primary"
+            onClick={(event) => {
+                event.preventDefault();
+                event.stopPropagation();
+                return onClick(event, handler, popperMsgKey);
+            }}
+            className={classes.actionButton}
+        >
+            {msg(name)}
+        </Button>
+    );
+};
+
+export default ({
+    row,
+    handleExtendTimeLimit = (_) => {},
+    handleDownloadInputs = (_) => {},
+    handleUploadOutputs = (_) => {},
+    handleExit = (_) => {},
+    handleSaveAndExit = (_) => {},
+}) => {
+    const classes = useStyles();
+
+    const [anchorEl, setAnchorEl] = useState(null);
+    const [open, setOpen] = useState(false);
+    const [popperMessage, setPopperMessage] = useState("");
+
+    const { status, data, error } = useQuery(
+        ["async-data", row.externalID],
+        asyncData
+    );
+
+    const isLoading = status === "loading";
+    const hasErrored = status === "error";
+
+    if (hasErrored) {
+        console.log(error);
```

Comment:
Took a little of bit of poking around, but I finally got the announcer to work here. I also changed things so that the action buttons don't get rendered if the request to get async data about the analysis fails. Also, each button can will cause an announcer to display, one with a message telling you that the action was successful or one to display an error message.